bundlegen/core/bundle_processor.py

```diff
@@ -28,17 +29,29 @@
 
 
 class BundleProcessor:
-    def __init__(self, platform_cfg, bundle_path, app_metadata, nodepwalking, libmatchingmode, createmountpoints):
-        self.platform_cfg: dict = platform_cfg
-        self.bundle_path = bundle_path
-        self.rootfs_path = os.path.join(self.bundle_path, "rootfs")
-        self.app_metadata = app_metadata
-        self.handled_libs = set()
-        self.createmountpoints = createmountpoints
-
-        self.oci_config: dict = self.load_config()
-        self.libmatcher = LibraryMatching(
-            self.platform_cfg, self.bundle_path, self._add_bind_mount, nodepwalking, libmatchingmode, createmountpoints)
+    def __init__(self, *args):
+        if (len(args)) == 6:
+            # Mapping of the arguments
+            # The arguments should be given in a order like (platform_cfg, bundle_path, app_metadata, nodepwalking, libmatchingmode, createmountpoints)
+            platform_cfg = args[0]
+            bundle_path = args[1]
+            app_metadata = args[2]
+            nodepwalking = args[3]
+            libmatchingmode = args[4]
+            createmountpoints = args[5]
+            self.platform_cfg: dict = platform_cfg
+            self.bundle_path = bundle_path
+            self.rootfs_path = os.path.join(self.bundle_path, "rootfs")
+            self.app_metadata = app_metadata
+            self.handled_libs = set()
+            self.createmountpoints = createmountpoints
+            self.oci_config: dict = self.load_config()
+            self.libmatcher = LibraryMatching(self.platform_cfg, self.bundle_path, self._add_bind_mount, nodepwalking, libmatchingmode, createmountpoints)
+        elif (len(args)) != 0:
+            logger.error("The arguments should be in a order like platform_cfg, bundle_path, app_metadata, nodepwalking, libmatchingmode, createmountpoints")
+            sys.exit(1)
```

Comment:
There must be a nicer way to handle this instead of just terminating the entire program here.

The `core` component of bundlegen should be treated as a library and should not impact lifecycle